File: Projects/Dagger/Projects/Project-2-FieldInjection/app/src/main/java/com/droid/code/di/modules/ActivityModule.java
package com.droid.code.di.modules;

import android.app.Activity;
import android.content.Context;

import com.droid.code.data.local.DatabaseService;
import com.droid.code.data.remote.NetworkService;
import com.droid.code.ui.MainViewModel;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {

    private Activity context;

    public ActivityModule(Activity context){
        this.context = context;
    }

    @Provides
    MainViewModel providesMainViewModule(DatabaseService databaseService,
                                         NetworkService networkService) {
        return new MainViewModel(databaseService,networkService);
    }

}

File: Projects/Dagger/Projects/Project-5-DaggerInKotlin/app/src/main/java/com/demo/kotlin/sample/di/module/ActivityModule.kt
package com.demo.kotlin.sample.di.module

import android.app.Activity
import android.content.Context

import com.demo.kotlin.sample.di.ActivityContext

import dagger.Module
import dagger.Provides

@Module
class ActivityModule(private val activity: Activity) {

    @ActivityContext
    @Provides
    fun provideContext(): Context = activity
}
